Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs

```diff
@@ -112,12 +112,12 @@ eval (UserProfile (GetProfileRes profile)) state = do
         _ -> Nothing
   continue state { data { name = name, editedName = name, gender = gender, emailId = profile.email} }
 eval (NameEditTextAction (PrimaryEditText.TextChanged id value)) state = do
-  if (value == "" || length (trim value) < 3) then continue state {data {editedName = value, nameErrorMessage = checkError "name" (Just state.data.name) value}, props{isNameValid = false, isBtnEnabled = ( (state.data.emailErrorMessage == Nothing) && ((checkError "name" (Just state.data.name) value) == Nothing) && (length (trim value) >=3) && state.props.isEmailValid && state.props.isNameValid && ((trim value) /= state.data.name) ) }}
-    else continue state { data { editedName = trim value, nameErrorMessage = Nothing }, props{isNameValid = true, isBtnEnabled = ((state.data.emailErrorMessage == Nothing) && ((checkError "name" (Just state.data.name) value) == Nothing) && (length value >=3) && state.props.isEmailValid && ((trim value) /= state.data.name) ||(state.data.emailId /= state.data.editedEmailId))} }
+  if (value == "" || length (trim value) < 3) then continue state {data {editedName = value, nameErrorMessage = checkError "name" (Just state.data.name) value}, props{isNameValid = false, isBtnEnabled = (state.data.emailErrorMessage == Nothing) &&  (length (trim value) >= 3) && ( state.props.isEmailValid && state.props.isNameValid && ((trim value) /= state.data.name) ||(state.data.emailId /= state.data.editedEmailId )) }}
+    else continue state { data { editedName = trim value, nameErrorMessage = Nothing }, props{isNameValid = true, isBtnEnabled = state.data.emailErrorMessage == Nothing &&  ( (length value >=3) && state.props.isEmailValid && ((trim value) /= state.data.name) ||(state.data.emailId /= state.data.editedEmailId))} }
```

Comment:
can we create `let` constants for most of the fields value instead of directly computing and storing inside state

Reply:
it would be easy to understand and debug